OpenDreamRuntime/Procs/DebugAdapter/DreamDebugManager.cs

```diff
@@ -325,23 +469,40 @@ private struct ActiveFunctionBreakpoint {
         reqThreads.Respond(client, threads);
     }
 
-    private void HandleRequestContinue(DebugAdapterClient client, RequestContinue reqContinue) {
-        Stopped = false;
-        stackFramesById.Clear();
-        variableReferences.Clear();
-        reqContinue.Respond(client, allThreadsContinued: true);
-    }
-
     private void HandleRequestPause(DebugAdapterClient client, RequestPause reqPause) {
         // "The debug adapter first sends the response and then a stopped event (with reason pause) after the thread has been paused successfully."
         reqPause.Respond(client);
-        Stop(new StoppedEvent {
+        Stop(null, new StoppedEvent {
             Reason = StoppedEvent.ReasonPause,
-            AllThreadsStopped = true,
             Description = "Paused by request",
         });
     }
 
+    private void HandleRequestContinue(DebugAdapterClient client, RequestContinue reqContinue) {
+        Resume();
+        reqContinue.Respond(client, allThreadsContinued: true);
+    }
+
+    private void HandleRequestStepIn(DebugAdapterClient client, RequestStepIn requestStepIn) {
+        threadStepModes[requestStepIn.Arguments.ThreadId] = (StepMode.StepIn, 0);
+        Resume();
+        requestStepIn.Respond(client);
+    }
+
+    private void HandleRequestNext(DebugAdapterClient client, RequestNext requestNext) {
+        var thread = InspectThreads().First(t => t.Id == requestNext.Arguments.ThreadId);
+        threadStepModes[requestNext.Arguments.ThreadId] = (StepMode.StepOver, thread.InspectStack().First().Id);
```

Comment:
The `First()` call here throws `System.InvalidOperationException: Sequence contains no elements` when the thread has nothing in its stack. `HandleRequestStepOut()` likely does this too.

This can most easily be replicated by pausing execution and trying to step the "Stat" thread outside of a `Stat()` call. That causes the exception and crashes the server.